src/cli.js

```diff
@@ -40,15 +39,32 @@ export default class Main {
     const plugins = new Plugins(out)
     await updater.autoupdate()
     await plugins.refreshLinkedPlugins()
-    let Command = plugins.findCommand(this.config.argv[1] || this.config.defaultCommand)
-    if (!Command) return new NotFound(out).run()
-    await out.done()
-    try {
+    if (this.cmdAskingForHelp()) {
+      this.cmd = await Help.run([this.cmdNeedingHelp()], this.config)
+    } else  {
+      let Command = plugins.findCommand(this.config.argv[1] || this.config.defaultCommand)
+      if (!Command) return new NotFound(out).run()
+      await out.done()
       this.cmd = await Command.run(this.config.argv.slice(2), this.config)
-    } catch (e) {
-      if (e instanceof HelpError) this.cmd = await Help.run([this.config.argv[1]], this.config)
-      else throw e
     }
     out.exit(0)
   }
+
+  cmdAskingForHelp () {
+    if (this.config.argv.includes('--')) return false
```

Comment:
Yeah, as discussed, we should allow (at least) `-h` to be a valid flag. I think we can hold off until another PR.